spec/prism_scanner_spec.rb

```diff
@@ -0,0 +1,420 @@
+          /Cyclic call detected: method_a -> method_b/
+        )
+      end
+
+      it 'returns nothing if only relative keys and private methods' do
+        source = <<~RUBY
+          class EventsController
+            private
+
+            def method_b
+              t('.relative_key')
+            end
+          end
+        RUBY
+
+        expect(
+          process_string('app/controllers/events_controller.rb', source)
+        ).to be_empty
+      end
+
+      it 'detects calls in methods' do
+        source = <<~RUBY
+          class EventsController
+            def create
+              t('.relative_key')
+              I18n.t("absolute_key")
+              method_b
+            end
+
+            def method_b
+              t('.error')
+              t("absolute_error")
+            end
+          end
+        RUBY
+
+        occurrences =
+          process_string('app/controllers/events_controller.rb', source)
+
+        expect(occurrences.map(&:first).uniq).to match_array(
+          %w[
+            absolute_key
+            absolute_error
+            events.create.relative_key
+            events.create.error
+            events.method_b.error
+          ]
+        )
+      end
+
+      it 'handles controller nested in modules' do
+        source = <<~RUBY
+          module Admin
+            class EventsController
+              def create
+                t('.relative_key')
+                I18n.t("absolute_key")
+                I18n.t(".relative_key_with_receiver")
+              end
+            end
+          end
+        RUBY
+
+        occurrences =
+          process_string('app/controllers/admin/events_controller.rb', source)
+
+        expect(occurrences.map(&:first).uniq).to match_array(
+          %w[
+            absolute_key
+            admin.events.create.relative_key
+            relative_key_with_receiver
+          ]
+        )
+      end
+
+      it 'handles controller with namespaced class name' do
+        source = <<~RUBY
+          class Admins::TestScopes::EventsController
+            def create
+              t('.relative_key')
+              I18n.t("absolute_key")
+            end
+          end
+        RUBY
+
+        occurrences =
+          process_string('app/controllers/admin/events_controller.rb', source)
+
+        expect(occurrences.map(&:first).uniq).to match_array(
+          %w[absolute_key admins.test_scopes.events.create.relative_key]
+        )
+      end
+
+      it 'rails model translations' do # rubocop:disable RSpec/MultipleExpectations
+        source = <<~RUBY
+          Event.human_attribute_name(:title)
+          Event.model_name.human(count: 2)
+          Event.model_name.human
+        RUBY
+
+        occurrences = process_string('app/models/event.rb', source)
+
+        expect(occurrences.map(&:first)).to match_array(
+          %w[
+            activerecord.attributes.event.title
+            activerecord.models.event.one
+            activerecord.models.event.other
```

Comment:
Maybe we would like some different way to return a `pluralized` translation, e.g. that it would just return `activerecord.models.event` with some `pluralized` setting and then this can be included in the tree.

Reply:
Yeah, I think that would be necessary to support pluralized keys properly.
Can be done in a follow-up PR